### PR TITLE
Support estatico newtypes

### LINKED ISSUE #56: Add support for the Newtype library
It seems the `@newtype` annotation from the [scala-newtype](https://github.com/estatico/scala-newtype) library doesn't work well with the `@derive` annotation. 

This fails to compile:

```scala
@newtype
@derive(show)
case class Foo(value: String)

@newtype
@derive(show)
case class Bar(value: Int)

@derive(show)
final case class Ctx(foo: Foo, bar: Bar)
```

Here's the error:

```
[error] /examples/foo.scala:19:4: magnolia: could not find Show.Typeclass for type examples..Foo
[error]     in parameter 'foo' of product type examples.Ctx
[error]   @derive(show)
[error]    ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
```

Maybe worth looking into [scalaz-deriving](https://github.com/scalaz/scalaz-deriving) that has support for it.

### PR DESCRIPTION
**Pull request by Odomontois**
fixes #56
Note that derevo-core dependency does not brings the estatico dependency.
Moreover, the mechanism could be easily extended for the "marks" of the other "newtype" libraries

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,11 +1,11 @@
 name := "derevo"
 import com.typesafe.sbt.SbtGit.git
 
-val publishVersion = "0.11.6"
+val publishVersion = "0.12.0"
 
 val common = List(
   scalaVersion := "2.13.4",
-  crossScalaVersions := List("2.12.12", "2.13.4"),
+  crossScalaVersions := List("2.12.13", "2.13.4"),
```
Comment (catostrophe) [inline build.sbt:8]: and here, please
https://github.com/tofu-tf/derevo/blob/3dfca42dcb172edc370124456faf19e8741cde2d/.github/workflows/scala.yml#L45

File: build.sbt [hunk 2]
```diff
@@ -21,6 +21,12 @@ val common = List(
...
   ),
+  Test / scalacOptions ++= Vector(
+    "-language:implicitConversions",
+  ),
+  libraryDependencies += "io.estatico"          %% "newtype"       % "0.4.4"            % Test,
+  libraryDependencies += "org.scalameta"        %% "munit"         % Version.munit      % "test",
+  libraryDependencies += "org.scalatest"        %% "scalatest"     % Version.scalaTest  % "test",
   scalacOptions ++= {
     CrossVersion.partialVersion(scalaVersion.value) match {
       case Some((2, y)) if y == 11 => Seq("-Xexperimental")
```
Comment (catostrophe) [inline build.sbt:32]: btw, not needed

File: build.sbt [hunk 3]
```diff
@@ -76,7 +82,7 @@ lazy val core = project settings common settings (
       s"""{"artifact": "$integrationOrg % ${integrationName}_2.13 % $integrationVersion"}""".stripMargin
     )
     Seq(intellijCompatFile)
-  }
+  },
```
Comment (FunFunFine) [inline build.sbt:85]:
```suggestion
  }
```

Comment (Odomontois) [reply build.sbt:84]: ?

File: core/src/main/scala/derevo/NewTypeRepr.scala [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package derevo
+
+//* utilitary holder for the newtype instance derivation */
```
Comment (FunFunFine) [inline core/src/main/scala/derevo/NewTypeRepr.scala:3]:
```suggestion
/* Utilitary holder for the newtype instance derivation */
```

File: build.sbt (hunk 3 above)
Comment (FunFunFine) [reply build.sbt:84]: This comma is not needed here

Comment (Odomontois) [reply build.sbt:84]: Idk, I like commas

**Comment by gvolpe** [comment]
That was quick! :rocket:

**Comment by gvolpe** [comment]
@Odomontois is there a SNAPSHOT I can try? I checked the CI actions, it seems there's no job publishing them automatically, no?

**Comment by Odomontois** [comment]
No, we haven't configured any snapshot/early yet

**Comment by gvolpe** [comment]
Ah no problems, no rush. Ping me when there's a new release / snapshot so I can keep up with my promise :smile:

**Comment by Odomontois** [comment]
@gvolpe we have released to sonatype https://oss.sonatype.org/content/groups/public/tf/tofu/
Waiting while the sync will be enabled with maven

**Comment by gvolpe** [comment]
Awesome, thanks for letting me know! :+1:

**Comment by gvolpe** [comment]
Bumped into this issue :crying_cat_face: https://github.com/tofu-tf/derevo/issues/220

**Pull request merged**